libs/cart/src/models/cart-item-input.ts

```diff
@@ -1,34 +1,75 @@
 import { ID } from '@daffodil/core';
 import { DaffProduct } from '@daffodil/product';
 
+/**
+ * An enum for the type of the cart item.
+ */
+// TODO: remove "input" designation
 export enum DaffCartItemInputType {
 	Simple = 'simple',
 	Composite = 'composite',
 	Configurable = 'configurable'
 }
 
+/**
+ * A set of fields specifying how to add a product to the cart.
+ */
 export interface DaffCartItemInput {
+  /**
+   * The type of cart item that correspondes to the product being added to the cart.
+   */
 	type: DaffCartItemInputType;
+  /**
+   * The ID of the product to add to the cart.
+   */
   productId: DaffProduct['id'];
+  /**
+   * How many of the product should be added to the cart.
+   */
 	qty: number;
 }
 
+/**
+ * A more constrained cart item input for simple products.
+ */
 export interface DaffSimpleCartItemInput extends DaffCartItemInput {
 	type: DaffCartItemInputType.Simple;
 }
 
+/**
+ * An cart item input for composite products.
+ */
 export interface DaffCompositeCartItemInput extends DaffCartItemInput {
 	type: DaffCartItemInputType.Composite;
+  /**
+   * The selected composite product options to be added to the cart.
+   */
 	options: DaffCompositeCartItemInputOption[];
 }
 
+/**
+ * Designates a particular composite product option to add to the cart.
+ */
 export interface DaffCompositeCartItemInputOption {
+  // TODO: what's this for? can we remove it?
 	code: string | number;
```

Comment:
For an example product like a toolbox composite product, the `code` could be a hammer that you could optionally add to the final product. Then the `value` would be like a sledge hammer or a claw hammer. So the code has to do with the parent "item" within the child "option".

Reply:
That being said, it might be possible to remove it, because options are always nested under items. I'm not sure if that would make our lives more difficult or not though

Reply:
Oh, nevemind. We definitely need it. I was thinking this was the `DaffCompositeProductItemOption`

Reply:
added some docs for it. I think it also makes sense to change it to the `ID` type. I don't like the `string | number` union